draft-ietf-taps-interface.md

```diff
@@ -989,6 +990,75 @@ Connection.Batch(
 )
 ~~~
 
+## Send on Active Open: InitiateAndSend {#initiate-and-send}
+
+For application-layer protocols where the Connection initiator also sends the
+first message, the InitiateAndSend() action combines Connection initiation with
+the first Message sent.
+
+Without a message context (as in {{send-basic}}):
+
+~~~
+Connection := Preconnection.InitiateAndSend(messageData)
+~~~
+
+With a message context (as in {{message-props}}):
+
+~~~
+Connection := Preconnection.InitiateAndSend(messageData, messageContext)
+~~~
+
+The message passed to InitiateAndSend is considered to be idempotent (see
+{{send-idempotent}}), regardless of declared message properties or defaults.
+If protocol stacks supporting 0-RTT establishment with idempotent data are
+available on the Preconnection, then 0-RTT establishment may be used with the
+given message when establishing candidate connections. For a non-idemponent
+initial message, or when the selected stack(s) do not support 0-RTT
+establishment, InitiateAndSend is identical to Initiate() followed by Send().
+
+Neither partial sends nor send batching are supported by InitiateAndSend().
+
+The Events that may be sent after InitiateAndSend() are equivalent to those
+that would be sent by an invocation of Initate() followed immediately by an
+invocation of Send(), with the caveat that a send failure that occurs because
+the Connection could not be established will not result in a
+SendError separate from the InitiateError signaling the failure of Connection
+establishment.
+
+## Send on Rendezvous: RendezvousAndSend {#rendezvous-and-send}
+
```

Comment:
I guess we need to think more about the rendezvous model. 

If the goal is to support traditional TCP simultaneous open, then 0-RTT idempotent data could work. For something like an ICE-driven rendezvous, with NAT traversal, I wonder if it still makes sense? STUN doesn't support delivering data in the binding discovery messages, so it's no longer 0-RTT. And, if some future protocol did support data in binding discovery messages, there's no guarantee that all those messages hit the same server, so the semantics get weird. 

I wonder if we should not try to support idempotent, 0-RTT, rendezvous for the first version?

Reply:
Sounds reasonable to leave it out for now.

Reply:
Yeah, I tried this out primarily to see what would happen if `RendezvousAndSend()` were kept consistent with `InitiateAndSend()` and you get this weird buffered-banner behavior that (1) probably doesn't do anything anyone wants and (2) also probably doesn't work anyway.

Will cut it for now, and we can leave rendezvous acceleration as an area for future research.